core/domain/stats_domain.py

```diff
@@ -1351,3 +1365,360 @@ def validate(self):
+                'received %s' % self.schema_version)
+
+        if not isinstance(self.accumulated_answer_info_json_size_bytes, int):
+            raise utils.ValidationError(
+                'Expected accumulated_answer_info_json_size_bytes to be an int '
+                'received %s' % self.accumulated_answer_info_json_size_bytes)
+
+    def add_learner_answer_info(self, learner_answer_info):
+        """Adds new learner answer info in the learner_answer_info_list.
+
+        Args:
+            learner_answer_info: LearnerAnswerInfo. The learner answer info
+                object, which is created after the learner has sbumitted the
+                details of the answer.
+        """
+        learner_answer_info_dict_size = (
+            learner_answer_info.get_learner_answer_info_dict_size())
+        if (self.accumulated_answer_info_json_size_bytes +
+                learner_answer_info_dict_size <= (
+                    MAX_LEARNER_ANSWER_INFO_LIST_BYTE_SIZE)):
+            self.learner_answer_info_list.append(learner_answer_info)
+            self.accumulated_answer_info_json_size_bytes += (
+                learner_answer_info_dict_size)
+
+    def delete_learner_answer_info(self, learner_answer_info_id):
+        """Delete the learner answer info from the learner_answer_info_list.
+
+        Args:
+            learner_answer_info_id: str. The learner answer info
+                id, which needs to be deleted from
+                the learner_answer_info_list.
+        """
+        deleted_learner_answer_info_id = (
+            learner_answer_info_id)
+        new_learner_answer_info_list = []
+        for learner_answer_info in self.learner_answer_info_list:
+            if learner_answer_info.id != deleted_learner_answer_info_id:
+                new_learner_answer_info_list.append(learner_answer_info)
+            else:
+                self.accumulated_answer_info_json_size_bytes -= (
+                    learner_answer_info.get_learner_answer_info_dict_size())
+        self.learner_answer_info_list = new_learner_answer_info_list
+
+    def update_state_reference(self, new_state_reference):
+        """Updates the state_reference of the LearnerAnswerDetails object.
+
+        Args:
+            new_state_reference: str. The new state reference of the
+                LearnerAnswerDetails.
+        """
+        self.state_reference = new_state_reference
+
+
+class LearnerAnswerInfo(object):
+    """Domain object containing the answer details submitted by the learner."""
+
+    def __init__(
+            self, learner_answer_info_id, answer,
+            answer_details, created_on):
+        """Constructs a LearnerAnswerInfo domain object.
+
+        Args:
+            learner_answer_info_id: str. The id of the LearnerAnswerInfo object.
+            answer: dict. The answer which is submitted by the learner,
+                actually type of the answer is interaction dependent but mostly
```

Comment:
learner. Actually the type of answer.....

Reply:
Done